iphone/Classes/TiAppiOSProxy.m

```diff
@@ -253,7 +257,7 @@ - (id)UserNotificationCenter
 
 - (void)didHandleURL:(NSNotification *)info
 {
-  if (![self _hasListeners:@"handleurl"]) {
+  if (![self _hasListeners:@"handleurl"] || !info.userInfo) {
```

Comment:
Same as above. In this case also userInfo always set. See [here](https://github.com/appcelerator/titanium_mobile/blob/2c2d7008f837da6b19c106414441cf7fe5ea9055/iphone/TitaniumKit/TitaniumKit/Sources/API/TiApp.m#L437).

Reply:
The issue is happening when the app is not booted. See the crash logs that definitely crash here. Using the fix it runs stable.

Reply:
As per code, if app is not booted launchOption is added in queuedBootEvents https://github.com/appcelerator/titanium_mobile/blob/2c2d7008f837da6b19c106414441cf7fe5ea9055/iphone/TitaniumKit/TitaniumKit/Sources/API/TiApp.m#L439 . And after app get booted this is used as userInfo in posting NSNotification as mentioned https://github.com/appcelerator/titanium_mobile/blob/2c2d7008f837da6b19c106414441cf7fe5ea9055/iphone/TitaniumKit/TitaniumKit/Sources/API/TiApp.m#L243 . So it should have userInfo always.

Can you attach crash log please?

Though adding extra condition will not create any problem. I am just thinking if there is any other issue.